app/controllers/application_controller.rb

```diff
@@ -1,5 +1,15 @@
-class ApplicationController < ActionController::API
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
-  # protect_from_forgery with: :exception
+class ApplicationController < ActionController::Base
+  protect_from_forgery with: :null_session
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def require_signin!
+    render status: 401,
+          json: {
+            success: false,
+            info: "Unauthorized"
+          } unless current_user
```

Comment:
Tacking this "unless" block onto the end of this json looks more like JS than Ruby to me. I think it reads fine, but is a little awkward because it appears after this longish block of json properties.